tools/isledecomp/isledecomp/parser/util.py

```diff
@@ -78,39 +51,45 @@ def is_blank_or_comment(line: str) -> bool:
     )
 
 
-def is_exact_offset_comment(line: str) -> bool:
-    """If the offset comment does not match our (unofficial) syntax
-    we may want to alert the user to fix it for style points."""
-    return offsetCommentExactRegex.match(line) is not None
-
-
-def match_offset_comment(line: str) -> OffsetMatch | None:
-    match = offsetCommentRegex.match(line)
+def match_marker(line: str) -> DecompMarker | None:
+    match = markerRegex.match(line)
     if match is None:
         return None
 
-    return OffsetMatch(
-        module=match.group(1),
-        address=int(match.group(2), 16),
-        is_template=match.group(3) is not None,
-        is_stub=match.group(4) is not None,
-        comment=line.strip(),
+    return DecompMarker(
+        type=match.group(1), module=match.group(2), offset=int(match.group(3), 16)
     )
 
 
-def distinct_by_module(offsets: List) -> List:
-    """Given a list of offset markers, return a list with distinct
-    module names. If module names (case-insensitive) are repeated,
-    choose the offset that appears first."""
+def is_marker_exact(line: str) -> bool:
+    return markerExactRegex.match(line) is not None
+
+
+template_class_decl_regex = re.compile(
+    r"\s*(?:\/\/)?\s*class (\w+)<([\w]+)\s*(\*+)?\s*>"
```

Comment:
Should we consider `struct` as an alternative to `class`? I don't think there's any cases where we use structs with vtables but maybe in the future

Reply:
I looked and didn't find any places where we use it right now. No problem to add it, though